includes/core/DataTransferObjects/API/class-response.php

```diff
@@ -0,0 +1,256 @@
+	 */
+	public function __construct( $body, $code, $headers, $message, $cookies ) {
+		$this->body          = $body;
+		$this->code          = $code;
+		$this->headers       = $headers;
+		$this->message       = $message;
+		$this->cookies       = $cookies;
+		$this->response_data = json_decode( $body, true );
+	}
+
+	/**
+	 * Returns true if response is JSON.
+	 *
+	 * @return bool
+	 */
+	public function is_json_response() {
+		return 'application/json' === substr( $this->headers['content-type'] ?? '', 0, strlen( 'application/json' ) );
+	}
+
+	/**
+	 * Get body.
+	 *
+	 * @return string
+	 */
+	public function get_body() {
+		return $this->body;
+	}
+
+	/**
+	 * Get statzs code.
+	 *
+	 * @return int
+	 */
+	public function get_code() {
+		return $this->code;
+	}
+
+	/**
+	 * Get headers.
+	 *
+	 * @return array
+	 */
+	public function get_headers() {
+		return $this->headers;
+	}
+
+	/**
+	 * Get message.
+	 *
+	 * @return string
+	 */
+	public function get_message() {
+		return $this->message;
+	}
+
+	/**
+	 * Get cookies
+	 *
+	 * @return WP_Http_Cookie[]
+	 */
+	public function get_cookies() {
+		return $this->cookies;
+	}
+
+	/**
+	 * Get response body as array.
+	 *
+	 * @return array|mixed
+	 */
+	public function get_response_data() {
+		$data = $this->response_data;
+		// Make sure empty metadata serialized on the client as an empty object {} rather than array [].
+		if ( isset( $data['metadata'] ) && empty( $data['metadata'] ) ) {
+			$data['metadata'] = new \stdClass();
+		}
+		return $data;
+	}
+
+	/**
+	 * Check if response is error one.
+	 *
+	 * @return bool
+	 */
+	public function is_error_response() {
+		return 400 <= $this->code;
+	}
+
+
+	/**
+	 * Get response message.
+	 *
+	 * @return mixed|null
+	 */
+	public function get_response_message() {
+		return $this->response_data['message'] ?? null;
+	}
+
+	/**
+	 * Get code from response data.
+	 *
+	 * @return mixed|null
+	 */
+	public function get_code_from_response_data() {
+		return $this->response_data['code'] ?? null;
+	}
+
+	/**
+	 * Get value from response by key.
+	 *
+	 * @param string $parameter Parameter in array.
+	 * @param string $key Key in array.
+	 * @return mixed|null
+	 */
+	public function get_value_from_response_by_key( $parameter, $key = 'data' ) {
+		return $this->response_data[ $key ][ $parameter ] ?? null;
+	}
+
+	/**
+	 * Get error code.
+	 *
+	 * @return mixed|null
+	 */
+	public function get_error_code() {
+		return $this->response_data['error']['code'] ?? null;
+	}
+
+	/**
+	 * Get error type
+	 *
+	 * @return mixed|null
+	 */
+	public function get_error_type() {
+		return $this->response_data['error']['type'] ?? null;
+	}
+
+	/**
+	 * Get error message.
+	 *
+	 * @return mixed|null
+	 */
+	public function get_error_message() {
+		return $this->response_data['error']['message'] ?? null;
+	}
+
+	/**
+	 * Determinate does code exists in response data.
+	 *
+	 * @return bool
+	 */
+	public function has_code_in_response_data() {
```

Comment:
I can easily remove all "syntax sugar" functions from the Response class, since at this phase and usage they will not be used. If we decide to step away from the current function that executes HTTP requests to WCPay server, the functions like these could help to become an easy way to get certain data somewhere in the code without doing manual checks.

If we would like to keep Response class clean and without too much code, I will remove all unused and unnecessary methods we don't use at this moment.

Reply:
> I can easily remove all "syntax sugar" functions from the Response class

I'm not sure what you mean by "syntax sugar" here, I only see a method, which is never used, and similar code in the API client class. If I saw this code being used instead, we could have worked with it, even though there would be similar code in the API client class. Otherwise we're left with two confusingly similar pieces of code, only one of which is ever used.

Reply:
This was just an easy way to get some data from the response object. Currently, we don't need functions like these, and I remove them.